lib/puppet/indirector/request.rb

```diff
@@ -250,8 +250,6 @@ def set_uri_key(key)
       return
     end
 
-    env, indirector, @key = URI.unescape(uri.path.sub(/^\//, '')).split('/',3)
-    @key ||= ''
-    self.environment = env unless env == ''
+    @key = key
```

Comment:
@joshcooper: you and I talked about this also. One idea was to throw an exception here since IIUC this should only happen if a new URI scheme is used.